core/templates/pages/preferences-page/preferences-page.component.ts

```diff
@@ -87,6 +91,24 @@ angular.module('oppia').component('preferencesPage', {
         _saveDataItem('user_bio', userBio);
       };
 
+      ctrl.checkBioChanged = function() {
+        ctrl.allowreload = false;
+      };
+
+      ctrl.saveBeforeUnload = function() {
+        WindowRef.nativeWindow.addEventListener(
```

Comment:
Should we only add this event listener when the content is changed? (And remove the listener once the changes are updated?)

Reply:
Let's keep the event listener for all so that if any other HTTP req does not get completed for another preference. It will give the warning (Here I'm keeping in mind about users who have slow internet)

Reply:
Sorry for the misunderstanding. I was asking: can we add the evntListner only when the bio is changed or https request is made? (And remove the listener once the HTTP request is complete?)

Reply:
We could try that. But, I don't understand the purpose of adding and removing the eventlistner?